sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawPod.java

```diff
@@ -79,6 +85,16 @@ private RawPod(
     this.sharePidNamespace = sharePidNamespace != null && sharePidNamespace;
     this.allowDecommission = allowDecommission != null && allowDecommission;
     this.hostVolumes = hostVolumes == null ? new WriteOnceLinkedHashMap<>() : hostVolumes;
+    this.seccompUnconfined = seccompUnconfined;
+    this.seccompProfileName = seccompProfileName;
+    validateSeccomp();
+  }
+
+  private void validateSeccomp() throws Exception{
+    if (seccompProfileName != null && seccompUnconfined != null) {
```

Comment:
this validation is incomplete...    It is necessary to validate not only that `seccompUnconfined` != null but also that if `seccompUnconfined` ==  true that `seccompProfileName` is null 

`if (seccompProfileName != null && (seccompUnconfined != null && seccompUnconfined)) {`

and we should have a test for this condition.